clang/lib/Sema/SemaInit.cpp

```diff
@@ -743,6 +743,12 @@ void InitListChecker::FillInEmptyInitForField(unsigned Init, FieldDecl *Field,
         ILE->updateInit(SemaRef.Context, Init, Filler);
       return;
     }
+
+    if (Field->hasAttr<ExplicitInitAttr>()) {
+      SemaRef.Diag(ILE->getExprLoc(), diag::warn_field_requires_explicit_init)
+          << Field;
```

Comment:
Maybe we could also add the `note_entity_declared_at` pointing at field's location?

It will mostly be redundant, but can sometimes really come in handy if the field names happen to clash with base classes, e.g. https://gcc.godbolt.org/z/Pzx9PE3Mh

However, this also adds noise.